### ISSUE TITLE
-Wunneeded-internal-declaration in drivers/platform/x86/mlx-platform.c

### ISSUE THREAD

**Issue by tpimh** [ClangBuiltLinux/linux#141]
```
drivers/platform/x86/mlx-platform.c:555:32: warning: variable 'mlxplat_mlxcpld_msn201x_items'
      is not needed and will not be emitted [-Wunneeded-internal-declaration]
static struct mlxreg_core_item mlxplat_mlxcpld_msn201x_items[] = {
                               ^
1 warning generated.
```

**Comment by nathanchance** [comment]
Patch accepted: http://git.infradead.org/linux-platform-drivers-x86.git/commit/8289c4b6f2e53750de78bd38cecb6bce4d7a988c

**Comment by nathanchance** [comment]
Merged into mainline: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=8289c4b6f2e53750de78bd38cecb6bce4d7a988c

**Comment by nickdesaulniers** [comment]
possibly also fixed on the llvm side by #232
